[doc] Clarify the nature of the root logger in the `logging` documentation

jkunimune opened the pull request:
The way it's currently written, it's very confusing what the root logger's name is, because it's not explicitly stated anywhere (the how-to page says that its name "is printed as 'root'", which makes it sound like it's actually something else, but it's not it's just 'root'), and because it is stated that the logging hierarchy is based on period-separated names ("root" is the exception to this rule, and that exception isn't stated anywhere here).

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--119440.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

jkunimune committed:
clarify details of root logger

The way it's currently written, it's very confusing what the root logger's name is, because it's not explicitly stated anywhere (the how-to page says that its name "is printed as 'root'", which makes it sound like it's actually something else, but it's not it's just 'root'), and because it is stated that the logging hierarchy is based on period-separated names (this is true except for "root", and that exception isn't stated anywhere here).

vsajip commented on Doc/library/logging.rst:
```diff
@@ -109,7 +109,8 @@ The ``name`` is potentially a period-separated hierarchical value, like
 higher up in the list.  For example, given a logger with a name of ``foo``,
 loggers with names of ``foo.bar``, ``foo.bar.baz``, and ``foo.bam`` are all
-descendants of ``foo``.  The logger name hierarchy is analogous to the Python
+descendants of ``foo``.  In addition, all loggers are descendants of ``root``.
```
```suggestion
descendants of ``foo``.  In addition, all loggers are descendants of the root logger.
```

vsajip commented on Doc/library/logging.rst:
```diff
@@ -1157,8 +1158,8 @@ functions.
-   Return a logger with the specified name or, if name is ``None``, return a
-   logger which is the root logger of the hierarchy. If specified, the name is
+   Return a logger with the specified name or, if name is ``None``, return the
+   root logger of the hierarchy (equivalent to passing ``name='root'``). If specified, the name is
```
I don't especially want to encourage the idiom where you fetch the root logger using `name='root'`. For example, code Python 2 doesn't work like this, and while Python 2.x is EOL, there are still environments where it is used (and there might be other environments like Jython where this isn't the case). So I'd prefer to suggest `logging.getLogger()` as the default way of getting the root logger, while perhaps noting that `logging.getLogger('')` or `logging.getLogger(None)` also work for this.

jkunimune committed:
Refer to the root logger consistently in the documentation

Understood about not wanting people to rely on the root logger's name being `'root'`.  I rephrased it to always refer to the root logger as simply "the root logger", as is done in the rest of the document.

vsajip commented on Doc/library/logging.rst:
```diff
@@ -1157,8 +1158,8 @@ functions.
+   Return a logger with the specified name or, if name is ``None``, return the
+   root logger of the hierarchy. If specified, the name is
    typically a dot-separated hierarchical name like *'a'*, *'a.b'* or *'a.b.c.d'*.
    Choice of these names is entirely up to the developer who is using logging.
```
```suggestion
   Choice of these names is entirely up to the developer who is using logging, though it is recommended that ``__name__`` be used unless you have a specific reason for not doing that, as mentioned in :ref:`logger-objects`.
```

vsajip reviewed (changes requested):
Your response to my earlier comment is fine, but as we're in the area, I thought maybe we can add another suggestion about logger naming. What do you think?

jkunimune committed:
Update Doc/library/logging.rst

Co-authored-by: Vinay Sajip <vinay_sajip@yahoo.co.uk>

vsajip committed:
Fix cross-reference and reformat paragraphs.
